sbi/inference/posteriors/sbi_posterior.py

```diff
@@ -456,36 +454,62 @@ def _x_else_x_o(self, x: Optional[Array]) -> Array:
         else:
             return self.x_o
 
+    def _process_x(self, x: Tensor) -> Tensor:
+        """Return x after casting to batched tensor and maybe comparing to x_o shape."""
+
+        x = atleast_2d(torch.as_tensor(x))
+        self._raise_error_for_multiple_x(x)
+
+        # XXX: This is to check the passed x matches the NN input dimensions by
```

Comment:
This is a great comment. Little nitpick: using TODO as flag seems to be a bit more featureful in VSCode.